src/collector/worker.rs

```diff
@@ -0,0 +1,150 @@
+    }
+    if connection.is_none() {
+        return None;
+    }
+    Some(RefMut::map(connection, |c| c.as_mut().unwrap()))
+}
+
+pub fn push(index_oid: u32, sample: &str, max_records: u32) {
+    let mut connection = match get() {
+        Some(c) => c,
+        None => return,
+    };
+    let init_statement = format!(
+        "
+        CREATE TABLE IF NOT EXISTS index_{index_oid} (id INTEGER PRIMARY KEY, sample TEXT, create_at REAL);
+        CREATE INDEX IF NOT EXISTS i ON index_{index_oid} (create_at DESC, id DESC);
+        "
+    );
+    let insert_statement = format!(
+        "INSERT INTO index_{index_oid} (sample, create_at) VALUES (?1, unixepoch('subsec'))"
+    );
+    let maintain_statement = format!(
+        "DELETE FROM index_{index_oid} WHERE (create_at, id) < (
+            SELECT create_at, id FROM index_{index_oid}
+            ORDER BY create_at DESC, id DESC
+            LIMIT 1 OFFSET ({max_records} - 1))"
+    );
+    let tx = match connection.transaction() {
+        Ok(t) => t,
+        Err(e) => {
+            pgrx::debug1!("Collector: Error starting transaction: {}", e);
+            return;
+        }
+    };
+    if let Err(e) = tx.execute_batch(&init_statement) {
+        pgrx::debug1!("Collector: Error creating table: {}", e);
+        return;
+    };
+    if let Err(e) = tx.execute(&insert_statement, (sample,)) {
+        pgrx::debug1!("Collector: Error inserting sample: {}", e);
+        return;
+    }
+    if let Err(e) = tx.execute(&maintain_statement, ()) {
+        pgrx::debug1!("Collector: Error maintaining samples: {}", e);
+        return;
+    }
+    let _ = tx.commit();
+}
+
+pub fn delete_index(index_oid: u32) {
+    let connection = match get() {
+        Some(c) => c,
+        None => return,
+    };
+    let drop_statement = format!("DROP TABLE IF EXISTS index_{index_oid}");
+    if let Err(e) = connection.execute(&drop_statement, ()) {
+        pgrx::debug1!("Collector: Error deleting index table: {}", e);
+    };
+}
+
+pub fn delete_database(database_oid: u32) {
+    if let Err(e) = fs::remove_file(format!("pg_vchord_sample/database_{database_oid}.sqlite")) {
+        pgrx::debug1!("Collector: Error deleting database file: {}", e);
+    };
+}
+
+pub fn dump(index_oid: u32, max_records: u32) -> Vec<String> {
+    let connection = match get() {
+        Some(c) => c,
+        None => return Vec::new(),
+    };
+    let load_statement =
+        format!("SELECT sample FROM index_{index_oid} ORDER BY create_at DESC limit {max_records}");
+    let mut stmt = match connection.prepare(&load_statement) {
+        Ok(s) => s,
+        Err(e) => {
+            pgrx::debug1!("Collector: Error preparing statement: {}", e);
+            return Vec::new();
+        }
+    };
+    let mut rows = match stmt.query(()) {
+        Ok(r) => r,
+        Err(e) => {
+            pgrx::debug1!("Collector: Error querying samples: {}", e);
+            return Vec::new();
+        }
+    };
```

Comment:
Extract the code into an immediately invoked closure, and ignore the returned `Err` or print logs.